app/assets/src/components/views/amr_heatmap/AMRHeatmapControls.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Divider } from "~/components/layout";
+import { Dropdown } from "~ui/controls/dropdowns";
+
+import cs from "./amr_heatmap_view.scss";
+
+export default class AMRHeatmapControls extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  generateChangeFunction(filter) {
+    const { selectedOptions, onSelectedOptionsChange } = this.props;
+    const changeFunction = option => {
+      if (option !== selectedOptions[filter]) {
+        onSelectedOptionsChange({ [filter]: option });
+      }
+    };
+    return changeFunction;
+  }
+
+  renderFilterDropdowns() {
```

Comment:
I really appreciate the effort to generalize functions and not duplicate code!

This is fine for now, but you'll probably need to generalize this further if we ever introduce a non-Dropdown control element.